src/backend/distributed/operations/shard_rebalancer.c

```diff
@@ -1971,16 +1972,45 @@ ExecuteRebalancerCommandInSeparateTransaction(char *command)
 	MultiConnection *connection = GetNodeConnection(connectionFlag, LocalHostName,
 													PostPortNumber);
 	StringInfo setApplicationName = makeStringInfo();
-	appendStringInfo(setApplicationName, "SET application_name TO %s",
+	appendStringInfo(setApplicationName, "SET application_name TO %s;",
 					 CITUS_REBALANCER_NAME);
 
+	StringInfo setStatements = GetSetStatementsForNewConnections();
+	appendStringInfoString(setApplicationName, setStatements->data);
+
 	ExecuteCriticalRemoteCommand(connection, setApplicationName->data);
 	ExecuteCriticalRemoteCommand(connection, command);
 
 	CloseConnection(connection);
 }
 
 
+StringInfo
+GetSetStatementsForNewConnections(void)
+{
+	StringInfo setStatements = makeStringInfo();
+
+	List *variableList = NIL;
+	char *splitCopy = pstrdup(VariablesToBePassedToNewConnections);
+
+	if (!SplitGUCList(splitCopy, ',', &variableList))
+	{
+		GUC_check_errdetail("not a valid list of identifiers");
+		return false;
+	}
```

Comment:
I would very much prefer if we can detect the variables that were changed at the session/transaction level automatically, instead of using a GUC to configure which variables get passed through.

I think this should be possible by using `get_guc_variables` and checking it's `GucSource`.